### PR TITLE
WIP: Adding tests to rollback

### PR DESCRIPTION
**Pull request by racs4**
For now, two test functions have been added, one capable of testing only a rollback (can be used easily for generating tests with proptests), and another capable of traversing a path of ticks and annotating selected points of interest (can also be used to generate tests, but tends to be much more expensive).
A simple checksum function was also made for the runtime equality check, which checks its state, size and mana.
The show_term function has been transformed from recursive to iterative for reasons of stack overflow.

### CODE REVIEW

File: src/hvm.rs [hunk 1]
```diff
@@ -2427,173 +2428,205 @@ pub fn show_rt(rt: &Runtime) -> String {
 }
 
 pub fn show_term(rt: &Runtime, term: Lnk) -> String {
-  let mut lets: HashMap<u128, u128> = HashMap::new();
-  let mut kinds: HashMap<u128, u128> = HashMap::new();
+  enum StackItem {
+    Term(Lnk),
+    Str(String),
+  }
```
Comment (VictorTaelin) [inline src/hvm.rs:2434]: Why a StackItem can be a Str?

**Comment by VictorTaelin** [comment]
Very good. Changing show_term to use a stacks is a great solution, but I still believe adding a `xorsum : Term -> u128` function is important: we could include that value on the block header in order to spot forks, solving the problem that we don't store the merkle root or any other kind of complete state hash.

**Comment by quleuber** [comment]
I agreed 100% the runtime state checksun is important, but I think we should think about that very carefully (I have some ideias) and treat this as completely separate from the ad-hoc checksum we are using for tests.

What I have in mind is that the protocol checksum has to operate on "incomplete" term trees (by that I mean they will have dangling/null children), as we have to only consider what has changed in the state at each tick, so a `xorsum : Term -> u128` is not even useful for that.

File: src/hvm.rs (hunk 1 above)
Comment (racs4) [reply src/hvm.rs:2434]: This way i can put a string in the stack. This is used to put spaces and closing parenthesis. 
Ex.: In an application: first I put an open parenthesis in the output; then i put a closing parenthesis, the second part of the application, a space and the first part of the application on the stack. In the next iterations the first part of the application will be resolved and will be placed in the output, followed by the space, the second part and the closing parenthesis.

**Comment by VictorTaelin** [comment]
Must resolve conflicts...

**Pull request merged**